_posts/2024-08-07-OpenSearch-for-Unified-Observability.md

```diff
@@ -0,0 +1,43 @@
+
+## Implications of Open Source
+
+Organizations today prefer open source, it’s the enabler of creativity and flexibility, and also something way more engineers know how to use when they join an organization. As the ecosystem evolves your tooling can evolve too, where you can have multiple offerings hopefully with different reasons to have each offering (cost, depth, complexity). Open source comes in many flavors, and understanding the ecosystem, health, and community are critical in picking the winning solutions. One such way to judge the viability of a project is to understand if the project is backed by a foundation and has a vibrant set of maintainers, committers, and community members to help you along with furthering the project. If the project lacks momentum, it can turn into a big cost when having to pick something new. The selection process along with the migration are painful. Getting stuck in a situation which is not ideal can often force this, including missing a bug fix, security patch, or even getting lack of response to an issue. 
+
+Data collection is much easier today thanks to OpenTelemetry, but it’s difficult to migrate the alerting, dash boarding, and overall user experience between tools. Moreover, it’s difficult to retrain users to learn new things, often having to build a user migration path to help users along. Open source makes this harder since this all rests on your shoulders as the decision maker for the platform. The users will feel the pain as well when you have to migrate instead of doing your day job. 
+
+OpenSearch provides a path forward which can be strategic, especially as the project moves towards being part of a foundation. This allows the project to have closer collaboration and alignment with an ecosystem of observability projects supported by the Cloud Native Computing Foundation (CNCF). 
+
+## Analysis using multiple interfaces
+
+OpenSearch is not only a database, but the powerful OpenSearch Dashboards allows the user to analyze observability data, namely logs and traces. In Jaeger (a project I am one of the maintainers of) we are working on some derived metrics on OpenSearch, but this is one weak spot for the database. I would not suggest doing high scale metrics on OpenSearch as It’s not suited for time series data. However, OpenSearch integrates with popular metric systems such as Prometheus, which is a better path forward for widespread use of metrics. 
+
+Although OpenSearch Dashboards is not as good as some other time series analysis tools such as Grafana, it doesn’t mean you can’t use both. Keep in mind that OpenSearch will never change licenses from its current Apache 2 license. This is not the case with other tools that are open source, but not part of software foundations such as the Apache Software Foundation, The Linux Foundation (LF) or the CNCF which is a sub-foundation of LF. 
```

Comment:
seems like a subjective comparative statement, can you help baseline or quantify ?

Reply:
This is my opinion based on in depth market understanding I'm happy to provide a detailed write-up of what's missing but I don't think this is a blog to compare OpenSearch to commerical software.